### ISSUE TITLE
Seasonal Adjustment error - Variable Date is not in domain

### ISSUE THREAD

**Issue by DG2DATA** [biolab/orange3-timeseries#281]
**What's wrong?**

Hi All,
I set the following steps but I have an error at the end (_Variable Date is not in domain_):

![image](https://github.com/biolab/orange3/assets/123603318/c030eaf7-8a7b-46ea-a238-9273fc56ec7e)

- **Details here:**

_Yahoo Finance_

![image](https://github.com/biolab/orange3/assets/123603318/09ce54f3-0b0f-4f02-a5af-36d5964b3b0c)

_Select Columns_

![image](https://github.com/biolab/orange3/assets/123603318/e31bf7e0-203c-4338-a8a2-f3619b88425c)

_Form Timeseries_

![image](https://github.com/biolab/orange3/assets/123603318/2e2cd2af-7def-4efa-9411-84ea0a64580b)

_Seasonal Adjustment_

![image](https://github.com/biolab/orange3/assets/123603318/b7a2a506-1aba-4a79-a209-a303edebe00f)

**How can we reproduce the problem?** 

Please see the steps reported above.

**What's your environment?**

- Operating system: Microsoft Windows 10 Pro
- Orange version: 3.37
- Timeseries add-on version: 0.6.3
- How you installed Orange: By using _Conda_

**Comment by jolo-maker** [comment]
Same issue with Seasonal Adjustment widget on OS X: 'Variable Month is not in domain'.
Orange version 3.36
TimeSeries-addon 0.6.3

![Scherm­afbeelding 2024-08-22 om 13 48 24](https://github.com/user-attachments/assets/e9cf4b65-9815-420d-9ee9-614683d6f60f)

**Comment by curry201205** [comment]
![image](https://github.com/user-attachments/assets/f21a4e88-8d69-468c-8feb-f04c4209dd7c)
after file widget, edit domain widget add and edit domain, this error not happend.
why?

**Comment by jolo-maker** [comment]
Inserting an Edit Domain widget does not help. Did you select one of the features for seasonal adjustment in the Seasonal Adjustment widget? Without a selection of a feature everything passes through unchanged, but we want to see the breaking down into trend, season and residual line plots right?
https://orangedatamining.com/widget-catalog/time-series/seasonal_adjustment/

**Comment by curry201205** [comment]
Edit Domain widget add, and change variable name .
 Then this error not happened on my case.
Why?

What's your environment?

- Operating system: Microsoft Windows 11 Pro
- Orange version: 3.37
- Timeseries add-on version: 0.6.3
- wsl (debian)

**Comment by curry201205** [comment]
Originally, I was reading in data without headers using the "File" widget and performing seasonal adjustment processing. The data consisted of two columns: one with the date and time recorded as features, and the other being the target data.
Since there were no headers, I changed the variable names using the "Edit Domain" widget partway through. At that time, I was able to perform seasonal adjustment without any issues.

However, when I tried to perform seasonal adjustment on data with column names added as headers, I encountered an error saying "variable ●● is not domain," and I was no longer able to perform the seasonal adjustment.

**Comment by jolo-maker** [comment]
Yes! Changing the time variable name in the Edit Domain does the trick! Unclear to me is why this is the case but the most important thing is that it works now, see picture.
Thanks!
![Scherm­afbeelding 2024-09-04 om 09 44 36](https://github.com/user-attachments/assets/45a104e5-08bd-474c-a3c0-edc8efe40aeb)

**Comment by DG2DATA** [comment]
Following your advices, the flow now works! Thanks, @curry201205  and @jolo-maker.
